### PR TITLE
fix  app to app calls in tproxy mode

### PR DESCRIPTION
**Pull request by ramaraochavali**
Fixes https://github.com/istio/istio/issues/31095

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ X] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by hzxuzhonghu — COMMENTED** [review]
lgtm

**Comment by ramaraochavali** [comment]
@hzxuzhonghu tests were failing with error "-o can not be specified with PREROUTING" - So I removed it but not sure of the implications though. Can you please check?

**Review by howardjohn — COMMENTED** [review]
can we add/extent the integration test to cover this?

File: tools/istio-iptables/pkg/cmd/run.go [hunk 1]
```diff
@@ -545,6 +545,9 @@ func (iptConfigurator *IptablesConfigurator) run() {
 	}
 
 	if iptConfigurator.cfg.InboundInterceptionMode == constants.TPROXY {
+		// rule to skip app ==> app traffic
+		iptConfigurator.iptables.InsertRuleV4(constants.PREROUTING, constants.MANGLE, 1, "-o", "lo", "-p", constants.TCP,
```
Comment (hzxuzhonghu) [inline tools/istio-iptables/pkg/cmd/run.go:549]: It is not -o lo, it is -i lo. I forgot which i sent u.

**Comment by hzxuzhonghu** [comment]
The test could be: enable strict mtls mode, and then make the app call itself by podip.

File: tools/istio-iptables/pkg/cmd/run.go (hunk 1 above)
Comment (ramaraochavali) [reply tools/istio-iptables/pkg/cmd/run.go:549]: Ah ok. Let me fix that.

**Comment by ramaraochavali** [comment]
@hzxuzhonghu I set it to -i - other tests are failing

File: tools/istio-iptables/pkg/cmd/run.go (hunk 1 above)
Comment (hzxuzhonghu) [reply tools/istio-iptables/pkg/cmd/run.go]: Let me debug

**Comment by ramaraochavali** [comment]
closing this infavour of https://github.com/istio/istio/pull/31165

**Pull request closed without merging**